toolkit/offchain/src/register.rs

```diff
@@ -0,0 +1,293 @@
+#![allow(dead_code)]
+
+use crate::csl::{InputsBuilderExt, TransactionBuilderExt, TransactionContext};
+use crate::plutus_script::PlutusScript;
+use cardano_serialization_lib::{
+	BigNum, DataCost, ExUnits, JsError, MinOutputAdaCalculator, Transaction, TransactionBuilder,
+	TransactionOutputBuilder, TxInputsBuilder,
+};
+use ogmios_client::types::OgmiosUtxo;
+use partner_chains_plutus_data::registered_candidates::block_producer_registration_to_plutus_data;
+use sidechain_domain::BlockProducerRegistration;
+
+pub fn register_tx(
```

Comment:
I don't think it should be public. Like with init_goveranance or d-param, there will be public higher level function, that calls this one after querying Cardano (via Ogmios) for the state.